.github/workflows/linux_job.yml

```diff
@@ -141,42 +144,91 @@ jobs:
+            p.stdin.write('set -e\n')
+            p.stdin.write(cmd)
+            p.stdin.write('\nexit $?\n')
+            p.stdin.close()
+
+            result = ''
+            while p.poll() is None:
+              line = p.stdout.readline()
+              if line:
+                print(line, end='')
+                result += line
+
+            # Read any remaining output
+            for line in p.stdout:
+              print(line, end='')
+              result += line
+
+            exit_code = p.returncode
+            if exit_code != 0:
+              raise RuntimeError(f'Command {cmd} failed with exit code {exit_code}')
+            return result
+
+          all_secrets = json.loads(os.environ['ALL_SECRETS'])
+          secrets_names = [x for x in '${{ inputs.secrets-env }}'.split(' ') if x]
+          if not secrets_names:
+            secrets_names = [x for x in all_secrets.keys()]
+          secrets_u_names = [
+            re.sub(r'[^a-zA-Z0-9_]', '', f'SECRET_{x.upper()}'.replace('-', '_'))
+            for x in secrets_names
+          ]
+
+          for sname, senv in zip(secrets_names, secrets_u_names):
+            try:
+              os.environ[senv] = str(all_secrets[sname])
+            except KeyError:
+              print(f'Could not set {senv} from secret {sname}')
+
+          container_name = run_cmd_or_die(f'''
+            docker run \
+              -e RUNNER_ARTIFACT_DIR=/artifacts \
+              -e RUNNER_TEST_RESULTS_DIR=/test-results \
+              --env-file="{ os.environ.get('RUNNER_TEMP', '') }/github_env_{ os.environ.get('GITHUB_RUN_ID', '') }" \
+              `# It is unknown why the container sees a different value for this.` \
+              -e GITHUB_STEP_SUMMARY \
+              { ' '.join([ f'-e {v}' for v in secrets_u_names ]) } \
+              --cap-add=SYS_PTRACE \
+              --detach \
+              --ipc=host \
+              --security-opt seccomp=unconfined \
+              --shm-size=2g \
+              --tty \
+              --ulimit stack=10485760:83886080 \
+              { os.environ.get('GPU_FLAG', '') if os.environ.get('CPU_FLAG') else '' } \
+              -v "{ os.environ.get('GITHUB_WORKSPACE', '') }/{ os.environ.get('REPOSITORY', '') }:/work" \
+              -v "{ os.environ.get('GITHUB_WORKSPACE', '') }/test-infra:/test-infra" \
+              -v "{ os.environ.get('RUNNER_ARTIFACT_DIR', '') }:/artifacts" \
+              -v "{ os.environ.get('RUNNER_TEST_RESULTS_DIR', '') }:/test-results" \
+              -v "{ os.environ.get('RUNNER_TEMP', '') }/exec_script:/exec" \
+              -v "{ os.environ.get('GITHUB_STEP_SUMMARY', '') }":"{ os.environ.get('GITHUB_STEP_SUMMARY', '') }" \
+              -w /work \
+              "{ os.environ.get('DOCKER_IMAGE', '') }"
+          ''').replace('\n', '').strip()
+          run_cmd_or_die(f'docker exec -t {container_name} /exec')
```

Comment:
Can we move this code to a separate .py file? We can also add tests to this logic.

Reply:
I am investigating this, to understand how reusable workflows deals with files from the current repo when called from another repo...

Reply:
We check out the test-infra repository in this job so anything you put in there should be fair game at `${GITHUB_WORKSPACE}/test-infra`